docs/zetacore/gas_and_fees.md

```diff
@@ -0,0 +1,334 @@
+# Gas and Fees in ZetaChain
+
+Gas and fees are fundamental mechanisms in blockchain networks.
+Gas represents the computational effort required to execute operations on a blockchain,
+while fees are the cost users pay for these resources,
+typically calculated by multiplying gas consumed by a gas price.
+Without fees, malicious actors could flood the network with spam transactions,
+overwhelming validators and halting the network.
+Fees make such attacks economically prohibitive,
+create efficient resource allocation through market mechanisms,
+and compensate validators for their computational work,
+ensuring the sustainability of the decentralized infrastructure.
+
+ZetaChain is a generalist interoperability protocol that enables execution of contracts on connected chains,
+including EVM and other virtual machine architectures.
+This creates multiple distinct contexts and situations where the notions of gas and fees apply differently.
+For example, the fee mechanism when directly interacting with the ZetaChain Layer 1 for a smart contract call differs from initiating a cross-chain transaction.
+This page summarizes all gas and fee mechanisms across ZetaChain's various components and connected chains.
+
+For simplicity, we will mostly refer to gas and fees as "fees" in the following sections.
+
+## Usages of Fees in ZetaChain
+
+ZetaChain involves fees in several distinct contexts:
+
+* **ZetaChain EVM Transaction Fees**: ZetaChain integrates the EVM module that makes it an EVM-compatible Layer 1 blockchain. Sending EVM transactions to ZetaChain involves using the standard EVM transaction fee mechanism.
+
+* **Connected Chains Withdraw Fees**: The calculation and payment of fees when a user initiates an outgoing cross-chain transaction from ZetaChain to a connected chain.
+
+* **Connected Chains Deposit Fees**: The calculation and payment of fees when a user initiates an incoming cross-chain transaction from a connected chain to ZetaChain.
+
+* **Cosmos Transaction Fees**: As a Layer 1 blockchain built with Cosmos SDK, sending direct Cosmos transactions on ZetaChain involves using the Cosmos gas mechanism for fee calculation and payment.
+
+## ZetaChain EVM Transaction Fees
+
+When users interact with the ZetaChain EVM, for any operations, they will pay fees using the standard EVM fee mechanism.
+More information can be found at the [EVM Fees Documentation](https://ethereum.org/developers/docs/gas/).
+
+ZetaChain currently uses Go-Ethereum `v1.15.11` and supports both legacy and EIP-1559 transaction types.
```

Comment:
Is it possible to pull this dynamically? It's one more thing we have to update each time we update cosmos-evm. 

If it'll take more than a few minutes to pull automatically then lets just skip it.